test/ranges.jl

```diff
@@ -168,9 +168,23 @@ end
             @test cmp_sn2(Tw(xw/yw), astuple(x/y)..., slopbits)
         end
     end
+    @testset "high precision of varying types" begin
+        x = Float32(π)
+        y = Float64(Base.MathConstants.γ)
+        @test Base.mul12(x, y)[1] ≈ Base.mul12(Float64(π), y)[1] rtol=1e-6
```

Comment:
Instead of testing each component, would it be better to test the values after calling `canonicalize2` on each output? I would also recommend testing against `Float64(x)` rather than `Float64(π)` since one expects much higher level of agreement.